### PR TITLE
feat: Add Nickel Language Server

### PR DESCRIPTION
**Pull request by Avi-D-coder**
Nickel Language Server
https://github.com/tweag/nickel

### CODE REVIEW

File: lua/lspconfig/nls.lua [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  Commands:
+
+  Default Values:
+    cmd = { "nls" }
+    filetypes = { "ncl", "nickel" }
+    root_dir = function(fname)
+          return util.find_git_ancestor(fname) or util.path.dirname(fname)
+        end,
+```
+        ]],
```
Comment (mjlbach) [inline lua/lspconfig/nls.lua:54]: Everything in the documentation past this line is (including this line) is auto-generated

File: lua/lspconfig/nls.lua [hunk 2]
```diff
@@ -0,0 +1,60 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+configs.nls = {
+  default_config = {
+    cmd = { 'nls' },
+    filetypes = { 'ncl', 'nickel' },
+    root_dir = function(fname)
+      return util.find_git_ancestor(fname) or util.path.dirname(fname)
```
Comment (mjlbach) [inline lua/lspconfig/nls.lua:9]: `root_dir = util.find_git_ancestor`, I'm not merging dirname hacks anymore

**Comment by mjlbach** [comment]
I think we should consider naming the config `nickel_ls` to avoid name collisions, `nls` is pretty generic.

**Comment by Avi-D-coder** [comment]
@mjlbach renamed

**Comment by mjlbach** [comment]
@regnat do you have any thoughts on naming, good with nickel_ls? Otherwise going to merge this :) (huge fan of your CAS work btw)

**Comment by Avi-D-coder** [comment]
nickel_ls seems good to me

**Comment by regnat** [comment]
> @regnat do you have any thoughts on naming, good with nickel_ls?

`nickel_ls` sounds good to me :+1:

**Comment by mjlbach** [comment]
Great :) We have nickel support now.

**Pull request merged**